### DISCUSSION TITLE
Should we release 0.73.5?

### DISCUSSION THREAD

**Discussion by lunaleaps** [reactwg/react-native-releases#103] [Patches]
Current Release: [0.73.4](https://github.com/facebook/react-native/releases/tag/v0.73.4)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

All qualified picks have been made. See latest[ status comment. ](https://github.com/reactwg/react-native-releases/discussions/103#discussioncomment-8516195)

**Comment by msageryd** [comment]
I'm not sure of how this works regarding branches. The commit I think should be picked is already merged into `main`, but I suppose `main` is not the same as the next release branch?

There are many issue-numbers involved and I don't know which to link here. I'll link the commit instead:
https://github.com/facebook/react-native/commit/c684f9fc62e865a5ecbd8ade0f5628f016eebcb2

This is a simple and risk-free fix that will make live much easier for those who use NVM or other Node version managers and does not have a regular Node installed.

**Reply by lunaleaps** [reply]
✅ picked

**Comment by cwhenderson20** [comment]
Can we pick https://github.com/facebook/hermes/commit/e7b2abefabb6a9671e1d30d7af08cd1f32c9a670 to fix a `WeakMap` memory leak in Hermes? Was confirmed safe for 0.73 [here](https://github.com/facebook/react-native/issues/42620#issuecomment-1906557596).

**Reply by lunaleaps** [reply]
✅ picked

**Comment by tido64** [comment]
There is a regression in 0.73 that prevents repos using pnpm from running any `react-native` command. We should cherry-pick the fix for it: https://github.com/facebook/react-native/commit/8f4f09bf704fbe5851d72bc5152ada0ab8c733b2

**Reply by lunaleaps** [reply]
We are aiming to release 0.73.5 on 02/20/2024

**Reply by lunaleaps** [reply]
✅ picked

**Comment by gituser8796** [comment]
Could you please include an important performance fix for Hermes JSON.stringify? @lunaleaps it was committed into main branch:

https://github.com/facebook/hermes/commit/628647010420d988ec4a08ec63742904c4bc785a

**Reply by lunaleaps** [reply]
✅ picked.

As an FYI, we ended up already committing the pick to the rn/0.73 branch from the `static-hermes` branch. I tried cp-ing this commit and there was no change. Instead of reverting and cp-ing this, I'm keeping the original pick and verified they are identical

**Comment by birdofpreyru** [comment]
Please pick https://github.com/facebook/react-native/commit/c684f9fc62e865a5ecbd8ade0f5628f016eebcb2

**Reply by lunaleaps** [reply]
This has already been requested.

**Comment by liamjones** [comment]
Is this something that'd be acceptable to be picked for a patch release or would it need to wait for 0.74? https://github.com/facebook/react-native/commit/7b7f128968310c466ce0a929e9964b08fe098503

**Reply by lunaleaps** [reply]
This is a new capability. It will be in 0.74

**Comment by tido64** [comment]
Codegen in 0.73 fails to find third-party libraries (both Yarn and pnpm setups are affected): https://github.com/facebook/react-native/pull/42943

This is already a PR on `0.73-stable`. The scripts have changed significantly on `main`.

**Reply by lunaleaps** [reply]
✅ merged

**Comment by lunaleaps** [comment]
Status update:

As of now, we won't be accepting any more pick requests. The outstanding issue is that the 0.73 Hermes branch has red CI and I'm following up internally to see how we should address.

Once I get an answer for that, will need to release a new Hermes tag and then `bump` the monorepo packages and test.

**Comment by cipolleschi** [comment]
Please pick:
* https://github.com/facebook/react-native/pull/43088
This is a pretty important fix as all the libraries that are using this function provided by React Native might suffer from problems when the New architecture is turned on.

**Comment by normtronics** [comment]
Please add: 
- https://github.com/facebook/react-native/pull/42639

After trying to upgrade our builds are now failing in CircleCi. See image below.
<img width="1280" alt="Screenshot 2024-02-19 at 4 22 47 PM" src="https://github.com/reactwg/react-native-releases/assets/2552632/852f6bf2-8566-4cd7-b04f-f1d06e0c12ef">

**Reply by lunaleaps** [reply]
I can't cleanly pick this -- @cipolleschi can you make a PR against 0.73 for this?

**Reply by lunaleaps** [reply]
@cipolleschi Here is how I ended up resolving: https://github.com/facebook/react-native/commit/0061bea215ef392c3e7fa5e7669bf25993cce490

**Reply by cipolleschi** [reply]
Thanks @lunaleaps for cherry picking this. The conflict resolution seems correct.

**Comment by fabOnReact** [comment]
Please pick 

https://github.com/facebook/react-native/commit/349d5502d0f0a7e8f66bedb3b8416dca37a41b76

Adding paragraphStyle.maximumLineHeight to a iOS UITextField displays the text under the UITextField ([ios-screenshot-1][1], [ios-screenshot-2][2], [ios-screenshot-3][3]). PR #38359 fixes the issue.

[1]: https://user-images.githubusercontent.com/24992535/238834159-566f7eef-ea2d-4fd4-a519-099b0a12046c.png "ios-screenshot-1"
[2]: https://user-images.githubusercontent.com/24992535/238834184-feb454a9-6504-4832-aec8-989f1d027861.png "ios-screenshot-2"
[3]: https://user-images.githubusercontent.com/24992535/238834283-cf572f94-a641-4790-92bf-bbe43afb1443.png "ios-screenshot-3"

**Reply by lunaleaps** [reply]
~~See https://github.com/reactwg/react-native-releases/discussions/103#discussioncomment-8516195~~

Okay I'll make the exception this case because we had other fixes to pick. Next time, please respect the pick deadline. We can try to be more clear about that.

**Comment by NickGerleman** [comment]
We should pick https://github.com/facebook/react-native/commit/e33767a36daf7901cb46854b6612dd52b0dbdeec to workaround a commonly reported Babel misconfiguration. 

Edit: just saw the comment about the pick deadline. If we end up doing another 0.73 release, I think this would meaningfully reduce issues reported and PRs trying to fix.

**Comment by efstathiosntonas** [comment]
Hi, `0.73.5` has landed on npm registry and upgrade helper but not on releases page. Just a heads up, thank you all for your hard work!

**Reply by lunaleaps** [reply]
Yup it's on Github releases now

**Comment by NathanNovak** [comment]
Source files are not available on an iOS device in experimental debugger when I upgraded to 0.73.4 or 0.73.5. I get the message.

`Failed to fetch source map http://192.168.86.64:8081/index.map?platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.AwesomeProject: remote fetches not permitted`

Can someone let me know what I am doing wrong?

**Reply by lunaleaps** [reply]
This is not the discussion space for issues for 0.73. Please file an issue on react-native repo instead
